### ISSUE TITLE
Better latency, error analysis navigation/overview

### ISSUE THREAD

**Issue by tarasglek** [openstatusHQ/openstatus#735]
### Description

I signed up for openstatus primarily for cool analytics capabilities per recent blog posts. Would love some improvements in this area.

1. I find it impossible to go from seeing red to finding response logs for red/failed responses.

<img width="1173" alt="image" src="https://github.com/openstatusHQ/openstatus/assets/857083/179c4304-f002-43f0-a731-1894aef9f2fc">

I think it would be intuitive to be able to click on failing indicators on main any dash that shows red and have those go to response view filtered specifically for failures. 

2. Likewise, it would be helpful to filter by p99, etc. Would love to see logs just for the fast/slow responses.

3. It would be helpful to adjust timeframe on /monitors page for 1 day, 7 days, month, etc.

5. Would be handy to download the database to do my own analytics

**Comment by mxkaske** [comment]
Hey @tarasglek, thanks a lot for the request!

A few thoughts before we start working on it (and we will!):

1. We will add a value filter value within the **Response Logs** to be able to filter by 'failed request' (true/false). That will provide an easy way to find the failed once. We are not using search params yet for the filters within a `/monitor/[id]/data` page but that might be the right time for doing it. That way, you can click on the failed requests and get forwarded to the correct filter option.
2. How do you want to filter by P99 exactly? The **Response Logs** are the 'raw' requests and so no aggregation has been done on it. Would you like to _only_ see the requests that are faster/slower than a specific threshold (or latency)? That's a more global approach where you define the max/min latency of your filter option.
3. You can adjust the timeframe on your monitor page (select option on top of the **uptime** card). That filter might not be obvious at all. It goes until 14d, but we'll provide more options. To avoid overloading the graph, we might restrict the 'Interval' to 1d for a month and higher. That's on the todo now!
4. Yes! We can provide you with a download button for the last X days. Is a `csv` file good?

Thanks for your time and making OpenStatus a better monitoring tool! ❤️

**Comment by tarasglek** [comment]
2. That's exactly what I want
3. Will check it out
4. Would prefer api access, eg if it could be exposed along with public status page that would be fine. This way you can skip auth.  csv download would work for now

**Comment by thibaultleouay** [comment]
Hey @tarasglek 

for 4 do you want a Prometheus endpoint?

**Comment by mxkaske** [comment]
I fully agree on (4): an API endpoint would be easier as you can access and process the data immediately, and always have up to date data.

**Comment by tarasglek** [comment]
@thibaultleouay i think both have usecases. For analytics a csv endpoint would be nice that i can plug into duckdb or something. For alerting and integrating into own infra prom endpoint would be better.

Actually do you mean endpoint that provides metrics or endpoint that I can query? I think both are useful :)

**Comment by thibaultleouay** [comment]
@tarasglek we have done a big design update of our platform I'm closing this
